recipes/dbg-macro/all/test_package/conanfile.py

```diff
@@ -11,6 +12,19 @@ def build(self):
         cmake.configure()
         cmake.build()
 
+    def configure(self):
```

Comment:
Now I see! YOu can't restrict anything in test package, you have to move it to the recipe package instead.

Reply:
But checking for compiler doesn't work for header-only recipes, does it?

I was getting an error:
```
ERROR: dbg-macro/0.4.0: Error in configure() method, line 24
	if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
	ConanException: 'settings.compiler' doesn't exist
'settings' possible configurations are none
```

Reply:
It works if you add to settings. As it's a header-only, the package ID can be unique, does not matter which settings are declared.

Reply:
Thanks.